src/topwidget.h

```diff
@@ -15,19 +15,24 @@ class TopWidget : public QToolBar
     virtual ~TopWidget();
 
     SearchBar &getSearchBar() { return m_searchEntry; };
+
+public slots:
+    void handleWebActionEnabledChanged(QWebEnginePage::WebAction action, bool enabled);
+    void updateBackForwardButtons();
+
 protected:
     void mousePressEvent(QMouseEvent *event);
     void mouseMoveEvent(QMouseEvent *event);
 
 private:
     SearchBar m_searchEntry;
-    QAction* mp_historyBackAction;
-    QAction* mp_historyForwardAction;
     QPoint m_cursorPos;
     ulong m_timestamp;
+    QMenu *back_menu = Q_NULLPTR;
+    QMenu *forward_menu = Q_NULLPTR;
```

Comment:
We should probably use a QScopedPointer (https://doc.qt.io/qt-5/qscopedpointer.html) here.
It would avoid us the `if (back_menu) { [delete] };` part (if we need it) and also automatically "delete" the pointer when we destruct the topWidget (and you forgot this part)

Reply:
Yes, good idea to illiminate this error-prone approach.
I use deleteLater() just for a case if page URL changed during history menu somehow is shown.  Hard to reproduce, but I think it's possible situation.